### PR TITLE
New UI

### PR DESCRIPTION
**Pull request by iGavroche**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,124 +1,194 @@
...
 if __name__ == "__main__":
-    gr.close_all()
-    launch_app()
\ No newline at end of file
+    import uvicorn
+
+    uvicorn.run("main:app", host="0.0.0.0", port=7865, reload=True)
+
+
+# import os
```
Comment (paolorechia) [inline app/main.py:128]: Any reason to keep old code commented?

File: app/memory/chroma_memory.py [hunk 2]
```diff
@@ -33,20 +42,25 @@ def __init__(self, **kwargs: Any):
...
+        collection = self.vector_store._client.get_collection(
+            self.collection_name
+        )
         if len(collection.get()['ids']) < 6:
-            self.add_texts(["Hello", "world!", "this", "is", "a", "test"], ids=[
-                           "1", "2", "3", "4", "5", "6"])
+            self.add_texts(
+                ["Hello", "world!", "this", "is", "a", "test"],
+                ids=["1", "2", "3", "4", "5", "6"],
+            )
```
Comment (paolorechia) [inline app/memory/chroma_memory.py:66]: Why do we have this block BTW? Could we remove it, or is it somehow still useful?

File: app/routes/conversations.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    if conversation_id is None:
+        raise HTTPException(status_code=404, detail="Conversation not found")
+    return retrieve_messages(conversation_id)
+
+
+@router.post("/conversations/{conversation_id}/messages")
+def add_message_to_conversation(conversation_id: str, message: str):
+    if conversation_id is None:
+        raise HTTPException(status_code=404, detail="Conversation not found")
+    save_messages(conversation_id, [message])
```
Comment (paolorechia) [inline app/routes/conversations.py:47]: Is this file actually used? At first glance it seems redundant with `main.py` - sorry if I missed something.

**Review by paolorechia — COMMENTED** [review]
Looks good to me in general, was just a bit confused about the conversation router, is it really used? Also still didn't read in detail the whole frontend code, overall it looks good though

File: app/main.py (hunk 1 above)
Comment (iGavroche) [reply app/main.py:128]: good catch!

File: app/memory/chroma_memory.py (hunk 2 above)
Comment (iGavroche) [reply app/memory/chroma_memory.py:66]: without it, when you query the index with `k=x` an error is raised if you do not have at least `x` documents in the vectordb. There is surely a better way to handle this. Let's table it for another PR.

File: app/routes/conversations.py (hunk 3 above)
Comment (iGavroche) [reply app/routes/conversations.py:47]: another great catch! No it's not anymore :)

**Review by paolorechia — APPROVED** [review]
Cool, Python wise looks good to me!

On the frontend I might not be the best reviewer though, I only had a bit of experience with Vue.

When looking at code structure it looks fine though, I can see we have some separation of concerns, like a router, an API layer etc.

Also really liked the icons :)

**Pull request merged**